File: src/server/gkill/api/req_res/download_skill_response.go
package req_res

import "github.com/mt3hr/gkill/src/server/gkill/api/message"

// DownloadSkillResponse はスキルの zip を base64 で返す（クライアントは JSON 以外の応答を受け付けないため）。
type DownloadSkillResponse struct {
	Messages  message.GkillMessages `json:"messages"`
	Errors    message.GkillErrors   `json:"errors"`
	FileName  string                `json:"file_name"`
	ZipBase64 string                `json:"zip_base64"`
}

File: src/client/classes/api/req_res/download-skill-response.ts
'use strict'

import { GkillAPIResponse } from '../gkill-api-response'

// DownloadSkillResponse はスキルの zip を base64 で返す（gkill_fetch は JSON 以外の応答を受け付けないため）。
// zip の中身はスキル名のフォルダ1段で包まれていて、そのままアップロードし直せる。
export class DownloadSkillResponse extends GkillAPIResponse {
    file_name: string
    zip_base64: string

    constructor() {
        super()
        this.file_name = ""
        this.zip_base64 = ""
    }
}
